ls1intum/Artemis
# Add TextSegmentationService. Integrate into TextClusteringService

- Add url to configs
- Add TextSegmentationService to call external service
- Add TextSegmentationServiceTest
- Integrate into TextClusteringService in compute Embeddings




### Checklist

### Motivation and Context
This change improves the segmentation of answers to textual exercises into text blocks.



### Description

A new service is added that runs in a virtual machine in a docker container alongside with Text-Assessment-Clustering. The code is in the same repository.

### Steps for Testing

1. ~Run the code form the Segmentation folder of the Text-Assessment-Clustering repository the same way as the Clustering code~ (The Test Server is configured to use our VM for that. :+1:)
1. Log in to Artemis
2. Navigate to Overview
3. Chose a course and a text exercise with more than 10 submissions. Submissions should contain reasonable text on the same topic (as in a real exercise). Not Lorem Ipsum.
4. Change the due date for the exercise to trigger the calculations
5. Inspect the resulting text blocks.
    - In the Database, OR
    - When Assessing.
